Secret-agency-web/src/main/java/cz/fi/muni/pa165/secretagency/security/AuthenticationProviderImpl.java

```diff
@@ -0,0 +1,58 @@
+package cz.fi.muni.pa165.secretagency.security;
+
+import cz.fi.muni.pa165.secretagency.dto.AgentAuthenticateDTO;
+import cz.fi.muni.pa165.secretagency.dto.AgentDTO;
+import cz.fi.muni.pa165.secretagency.enums.AgentRankEnum;
+import cz.fi.muni.pa165.secretagency.facade.AgentFacade;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class AuthenticationProviderImpl implements AuthenticationProvider {
+
+    @Autowired
+    private AgentFacade agentFacade;
+
+    @Override
+    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+        String codeName = authentication.getName();
+        String password = (String) authentication.getCredentials();
+
+        // TODO - create custom exception
```

Comment:
mas v planu s tim neco delat? 😄

Reply:
Momentálně ne. 😄 To je na dobu, až bude čas zabývat se drobnostma. 😄